src/Redux/missions/missionsSlice.js

```diff
@@ -0,0 +1,65 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const apiUrl = 'https://api.spacexdata.com/v3/missions';
+
+export const fetchMissions = createAsyncThunk('missions/fetchMissions', async () => {
+  const response = await axios.get(apiUrl);
+  return response.data;
+});
+
+const initialState = {
+  missions: [],
+  loading: false,
+  error: null,
+  reservedMissions: [],
+};
+
+const missionSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    joinMission: (state, { payload }) => {
+      state.reservedMissions.push(payload);
```

Comment:
- [ ] I will encourage you not to mutate the state, kindly use spread operator instead, you can readup on it for more details on why it is unprofessional.